File: .tmp_py4013/batch_09_dump/3012_minimize_length_of_array_using_operations.js
// LeetCode 3012 - Minimize Length of Array Using Operations
// https://leetcode.com/problems/minimize-length-of-array-using-operations/

var minimumArrayLength = function(nums) {
    let mi = nums[0];
    for (const x of nums) if (x < mi) mi = x;
    let cnt = 0;
    for (const x of nums) {
        if (x % mi !== 0) return 1;
        if (x === mi) cnt++;
    }
    return ((cnt + 1) / 2) | 0;
};

File: .tmp_rs4013/rust_check13/3012_minimize_length_of_array_using_operations.rs
#![allow(dead_code, unused_variables, unused_mut, unused_assignments, unused_imports)]
struct Solution;
// LeetCode 3012 - Minimize Length of Array Using Operations
// https://leetcode.com/problems/minimize-length-of-array-using-operations/

impl Solution {
    pub fn minimum_array_length(nums: Vec<i32>) -> i32 {
        let mi = *nums.iter().min().unwrap();
        let mut cnt = 0;
        for &x in &nums {
            if x % mi != 0 {
                return 1;
            }
            if x == mi {
                cnt += 1;
            }
        }
        (cnt + 1) / 2
    }
}
